src/main/java/org/broadinstitute/hellbender/utils/config/ConfigUtils.java

```diff
@@ -0,0 +1,332 @@
+        // NOTE: The configuration will be stored in the ConfigCache under the key GATKConfig.class.
+        //       This means that any future call to getOrCreate for this GATKConfig.class will return
+        //       Not only the configuration itself, but also the overrides as specified in userConfigFileProperties
+        return ConfigCache.getOrCreate(configClass, userConfigFileProperties);
+    }
+
+    /**
+     * Injects system properties from the given configuration file.
+     * @param config The {@link GATKConfig} object from which to inject system properties.
+     */
+    public static final <T extends Config> void injectSystemPropertiesFromConfig(T config) {
+        injectSystemPropertiesFromConfig(config, "");
+    }
+
+    /**
+     * Injects system properties from the given configuration file.
+     * @param config The {@link GATKConfig} object from which to inject system properties.
+     * @param propertyPrefix Prefix that is required for a property in {@code config} to be imported to System properties.
+     */
+    public static final <T extends Config> void injectSystemPropertiesFromConfig(T config, String propertyPrefix) {
+        
+        Utils.nonNull(config);
+        Utils.nonNull(propertyPrefix);
+
+        // This is gross and uses reflection to get all methods in the given config class
+        // and then interrogates those methods for internal data on the config parameters.
+
+        // We have to match our interfaces to the config interface that we're actually using.
+        // It's not as simple as using getDeclaredMethods on the Class object because we'll get
+        // a LOT of extraneous stuff that we don't care about.
+        for ( Class<?> classInterface : config.getClass().getInterfaces() ){
+
+            // If we have an interface that is a child of the OWNER Config interface, then
+            // we must have the right object.
+            if (Config.class.isAssignableFrom(classInterface)) {
+
+                // Now we cycle through our interface methods, resolve parameter names,
+                // and set the values in the system.
+                for (Method propertyMethod : classInterface.getDeclaredMethods()) {
+
+                    // Get the property name:
+                    String propertyName = propertyMethod.getName();
+
+                    Config.Key key = propertyMethod.getAnnotation(Config.Key.class);
+                    if (key != null) {
+                        propertyName = key.value();
+                    }
+
+                    // Only set properties beginning with the given prefix:
+                    if (!propertyName.startsWith(propertyPrefix)) {
+                        continue;
+                    }
+
+                    // Get the value of the property into a stringbuilder:
+                    StringBuilder sb = new StringBuilder();
+                    try {
+                        sb.append(propertyMethod.invoke( config, new Object[]{} ));
```

Comment:
Instead of using a StringBuilder I think you should either just call `String.valueOf()` or do an explicit cast if you know that's safe.